### PR TITLE
Njavokhir/new world transition #21

### PR DESCRIPTION
**Pull request by NJavokhir**
#### **Summary:**
This PR addresses and closes issue #21 by implementing a smoother fade-out animation for the raining effect and introducing a progress-loading widget during chapter transitions.

#### **Key Changes:**
1. **Smooth Fade-out for Raining Effect:**
   - Updated the `stopRainingEffect`, `startRainingEffect` to smoothly fade out the canvas over 1 second before removing it.
   
   Issue #21 is now resolved with these updates.

### CODE REVIEW

**Comment by stravid87** [comment]
Good to see progress. There are still some problems here.

File: src/utils/codeRainingEffect.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+let intervalId: number | null = null;
+let canvas: HTMLCanvasElement | null = null;
+let style: HTMLStyleElement | null = null;
+
+export const startRainingEffect = () => {
+  if (intervalId) return;
+
+  canvas = document.createElement('canvas');
+  canvas.style.opacity = '0';
+  canvas.style.transition = 'opacity 1s ease';
```
Comment (stravid87) [inline src/utils/codeRainingEffect.ts:10]: Here you are using CSS styles. However, on line 95 of `src/views/HomeView.vue`, you again are using transitions. Are both necessary? Perhaps one or the other?

File: src/utils/codeRainingEffect.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      width: 100%; 
+      position: absolute; 
+      top: 0; 
+      left: 0; 
+      z-index: 9999;
+    }
+  `;
+  document.head.appendChild(style);
+
+  if (canvas) {
```
Comment (stravid87) [inline src/utils/codeRainingEffect.ts:28]: From here, Line 28 to Line 59 is your main function that creates the raining code effect. Everything above and below is code you'll write to have it transition appropriately. I notice a lot of timers. Try to architect a more graceful solution that takes advantage of asynchronous calls, callbacks, or other mechanisms to make the transitions between chapters smooth and predictable.

File: src/utils/codeRainingEffect.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+export const stopRainingEffect = () => {
+  if (intervalId) {
+    clearInterval(intervalId);
+    intervalId = null;
+  }
+
+  if (canvas) {
+    canvas.style.transition = 'opacity 1s ease'; 
+    canvas.style.opacity = '0';
```
Comment (stravid87) [inline src/utils/codeRainingEffect.ts:75]: I am not seeing a slow fade out.

File: src/utils/codeRainingEffect.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    setTimeout(() => {
+      if (canvas) {
+        document.body.removeChild(canvas);
+        canvas = null;
+      }
+    }, 1000);
```
Comment (stravid87) [inline src/utils/codeRainingEffect.ts:82]: This causes the effect to disappear abruptly. It should fade out.

File: src/views/HomeView.vue [hunk 5]
```diff
@@ -72,6 +86,16 @@ onMounted(() => {
     </div>
     <FooterComponent></FooterComponent>
   </main>
+  <div v-if="isRaining" class="fade-in" @animationend="stopRainingEffect"></div>
 </template>
 
-<style lang="scss"></style>
+<style lang="scss">
```
Comment (stravid87) [inline src/views/HomeView.vue:92]: Are we using Scss? Copy/paste is okay if it get cleaned up. I feel like this has been left behind.

**Review by stravid87 — CHANGES_REQUESTED** [review]
You should open your PRs against, `njavokhir-base-branch`, not develop.
Some progress -- but we're not quite there yet.

**Comment by NJavokhir** [comment]
I looked your feedbacks and do some changes related to them. I am going to close this PR because I opened it for `develop` branch. New PR has already opened for `njavokhir-base-branch` which is #41 .

**Pull request closed without merging**